### PR TITLE
LPS-113561 Avoid use of Liferay.provide in users-admin-web

### PR DESCRIPTION
**Pull request by wincent**
As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561) we 
eventually want to deprecate the `Liferay.provide` function

To test this change:
- From the "Global Menu" select the "Control Panel" tab
- Click on "Users and Organizations"

You should be able to:

- Create new users 
- Assign roles to users
- Create organizations
- Add users to organizations

### CODE REVIEW

**Comment by wincent** [comment]
Will want to re-review this, seeing as I haven't looked at it since the last changes on https://github.com/wincent/liferay-portal/pull/337 — but in the meantime, re-trying the tests.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
@julien: We might still need a squash/push of this one, because I see:

- [Suggestion to squash because history is confusing](https://github.com/wincent/liferay-portal/pull/337#issuecomment-642009052).
- [Report that everything is squashed](https://github.com/wincent/liferay-portal/pull/337#issuecomment-642011673).
- [Comment that it looks like squahsed changes not pushed](https://github.com/wincent/liferay-portal/pull/337#issuecomment-642070111).

So maybe that comment got overlooked and the squash/push never made it here. Or maybe more commits got added after that conversation (hard to see in the GitHub history, because I see a "added 6 commits" annotation, and also a later "force push" one). At the moment the commit messages read like this (where the repeated explanation and test plan makes it difficult for me to see what actually happened):

```
commit 09f6d4d6b8a6684af8ddc629fe62314700332853
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Mon Jun 8 12:12:50 2020 +0200

    LPS-113561 Avoid declaring `<portlet:namespace />selectRole` function as global
    
    As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561) we
    eventually want to deprecate the `Liferay.provide` function.
    
    In this change, we avoid declaring the `<portlet:namespace />selectRole`
    function on the global `window` object and remove the unnecessary
    `liferay-search-container` AUI dependency.
    
    To test this change:
    
    - From the "Global Menu" select the "Control Panel" tab
    - Click on "Users and Organizations"
    - Click on the "+" button to create a new user
    - Enter a screen name (for example User01)
    - Enter an email address (for example user01@liferay.com)
    - Enter a value for First Name (for example User)
    - Enter a value for Last Name (for example 01)
    - Click on the "Save" button
    - Select the "Roles" tab
    - Click on the "Select" button under the "Regular Roles" section
    - Choose a role in the "Select Regular Role" modal window
    
    As a result the role you selected should be visible in the "Regular
    Roles" section and no JS errors should appear in the console.

commit e9e833417a004cdb02b91f22557fab7ff604176f
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Mon Jun 8 12:41:44 2020 +0200

    LPS-113561 Avoid declaring `openSelectUsersDialog` as a global
    
    As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561)
    eventually we want to deprecate the `Liferay.provide` function.
    
    In this change, we avoid declaring the `<portlet:namespace/
    >openSelectUsersDialog` function on the global `window` object.
    
    To test this change:
    - From the "Global Menu" select the "Control Panel" tab
    - Click on "Users and Organizations"
    - Select the "Organizations" tab
    - Click on the "+" button to create a new organization
    - Enter a name (for example Organization01)
    - Click on the "Save" button
    - Navigate back the organizations list
    - Click on the kebab menu on the right of the organization
    - Click on the "Assign Users" menu item
    - Select one or more users and click on the "Done" button
    
    As a result you should be redirected to the new organization's page and
    see the users that are part of this organization.

commit 28b8f01abf175438c2791edcec19974489f8bb1f
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Tue Jun 9 08:26:22 2020 +0200

    LPS-113561 Avoid unnecessary use of `Liferay.provide` in users-admin-web
    
    As commented [here](https://github.com/wincent/liferay-portal/pull/320#discussion_r436726226),
    the use of `Liferay.provide` is not necessary and
    can be removed.
    
    Additionally, instead of using the `<portlet:namespace />` as a prefix
    for the function name is also not required.
    
    To test this change:
    
    - From the "Global Menu" select the "Control Panel" tab
    - Click on "Users and Organizations"
    - Click on the "+" button to create a new user
    - Enter a screen name (for example User01)
    - Enter an email address (for example user01@liferay.com)
    - Enter a value for First Name (for example User)
    - Enter a value for Last Name (for example 01)
    - Click on the "Save" button
    - Select the "Roles" tab
    - Click on the "Select" button under the "Regular Roles" section
    - Choose a role in the "Select Regular Role" modal window
    
    As a result the role you selected should be visible in the "Regular
    Roles" section and no JS errors should appear in the console.

commit 3cd1772323b6dc98abea25775b7f3236975bc74a
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Tue Jun 9 08:37:43 2020 +0200

    LPS-113561 Avoid unnecessary use of `Liferay.provide` in users-admin-web
    
    As commented [here](https://github.com/wincent/liferay-portal/pull/320#discussion_r436726226),
    the use of `Liferay.provide` is not necessary in this case and can be removed.
    Additionally, the functions don't need to be prefixed with `<portlet:namespace />`.
    
    To test this change:
    - From the "Global Menu" select the "Control Panel" tab
    - Click on "Users and Organizations"
    - Select the "Organizations" tab
    - Click on the "+" button to create a new organization
    - Enter a name (for example Organization01)
    - Click on the "Save" button
    - Navigate back the organizations list
    - Click on the kebab menu on the right of the organization
    - Click on the "Assign Users" menu item
    - Select one or more users and click on the "Done" button
    
    As a result you should be redirected to the new organization's page and
    see the users that are part of this organization.

commit 41bffdc63b35c640d27b35a110bdae343e326381
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Wed Jun 10 09:18:00 2020 +0200

    LPS-113561 Don't expose a `Liferay.Users` API

commit ffb7a91d8aa42639254e6b413191e98998deacac
Author: Julien Castelain <julien.castelain@liferay.com>
Date:   Fri Jun 12 13:44:28 2020 +0200

    LPS-113561 Add 'liferay-search-container' to <aui:script> tags
```

I'll leave it up to you about whether to change that. I'll just review the code and not pay too much attention to the messages.

**Comment by julien** [comment]
@wincent, sorry I'm gonna squash all those commits.

**Comment by julien** [comment]
ci:stop

**Comment by julien** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Without looking in detail at the contents of the failure, the fact that it is in `LocalFile.WorkflowUser#CreateNewUserAccount` does look suspicious. Could you please confirm @john-co that this is related? I see it's failing to find `//a[(contains(@class,'active')) and (contains(text(),'Assigned to Me'))]` and while I can't see explicitly where we are setting or not setting that class in this PR, I do see references to the word `active` in the source.

cc @julien — might have to look into this tomorrow: assigning to you just in case.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp [hunk 1]
```diff
@@ -95,257 +95,257 @@ else {
...
+		'<liferay-portlet:resourceURL id="/users_admin/get_users_count" />',
+		{
+			body: formData,
+			method: 'POST',
+		}
+	)
+		.then(function (response) {
+			return response.text();
+		})
+		.then(function (response) {
```
Comment (wincent) [inline modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp:257]: Every `then()` should have a `catch()`.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp [hunk 2]
```diff
@@ -95,257 +95,257 @@ else {
...
+	showUsersURL.setParameter("toolbarItem", toolbarItem);
+	showUsersURL.setParameter("usersListView", usersListView);
 
-		showUsersURL.setParameter("mvcRenderCommandName", "/users_admin/view");
-		showUsersURL.setParameter("toolbarItem", toolbarItem);
-		showUsersURL.setParameter("usersListView", usersListView);
+	organizationId = ParamUtil.getLong(request, "organizationId", OrganizationConstants.DEFAULT_PARENT_ORGANIZATION_ID);
 
-		organizationId = ParamUtil.getLong(request, "organizationId", OrganizationConstants.DEFAULT_PARENT_ORGANIZATION_ID);
+	if (organizationId != OrganizationConstants.DEFAULT_PARENT_ORGANIZATION_ID) {
```
Comment (wincent) [inline modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp:273]: Ha, I was going to suggest `!==` here and then realized it was Java. 😂

There *aren* some `==` elsewhere in this PR that would be better changed to `===`; every little step towards being able to turn on the `eqeqeq` ESLint rule helps... Up to you whether you want to do it or not (totally optional!).

**Comment by wincent** [comment]
Code changes look fine to me @julien, other than the noted thing about the `catch()`, and the optional `==` → `===`.

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Hi @wincent, It looks unrelated, there's a related ticket that was recently fixed: https://issues.liferay.com/browse/LRQA-59457

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp (hunk 1 above)
Comment (julien) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp:257]: I could and maybe should have added it. I usually just replace the AUI functions, and leave the rest as is but I'll replace it.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp (hunk 2 above)
Comment (julien) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view.jsp:273]: Thanks for the heads up - I'm going to replace those

**Comment by julien** [comment]
@wincent apart from applying the changes you suggested, I updated the description

**Comment by wincent** [comment]
Ok, let's try forwarding this.

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**